nistats/first_level_model.py

```diff
@@ -557,3 +568,214 @@ def compute_contrast(self, contrast_def, stat_type=None,
             '%s of contrast %s' % (output_type, contrast_name))
 
         return output
+
+
+def first_level_models_from_bids(dataset_path, task_id, model_init=None,
+                                 preproc_variant=None, preproc_space=None):
+    """Create FirstLevelModel objects and fit arguments from a BIDS dataset.
+
+    Parameters
+    ----------
+    dataset_path: str
+        Directory of the highest level folder of the BIDS dataset. Should
+        contain subject folders and a derivatives folder.
+
+    task_id: str
+        Task_id as specified in the file names. Of the form _task-*_.
+
+    model_init: dict, optional
+        Is posible to specify the parameters of the FirstLevelModel objects
+        as a dictionary. The same parameter initialization will be applied to
+        all FirstLevelModel objects.
+
+    preproc_variant: str, optional
+        Can specify a particular variant label of the preproc.nii images.
+        As they are specified in the file names.
+
+    preproc_space: str, optional
+        Can specify a particular space label of the preproc.nii images.
+        As they are specified in the file names.
+
+    Returns
+    -------
+    models: list of `FirstLevelModel` objects
+        Each FirstLevelModel object corresponds to a subject. All runs from
+        different sessions are considered together for the same subject to run
+        a fixed effects analysis on them.
+
+    models_fit_kwargs: list of dict
```

Comment:
I thing that this overly generic argument is bad design. Imho we need to enforce much more structure in it.